### ISSUE TITLE
scheduling: redundant triggers pass strict validation

### ISSUE THREAD

**Issue by sadielbartholomew** [cylc/cylc-flow#2914]
When trigger syntax e.g. ``:fail`` is appended to any task that itself is not a dependency to other task(s) i.e. that is a graph string endpoint, these triggers are not picked up.

While such triggers are clearly ineffectual, as in isolation they have no effect on the scheduling graph, they could possibly (?) be harmless in some way, perhaps when included in a more involved graph string utilising other features e.g. families, or in terms of security, given any non-functional text of alphanumeric characters & e.g. hyphens can be inserted after a colon at the end of a valid task name (see below)? I think they at least should be registered as redundant as a warning in ``cylc validate``, since I believe it is possible to pick up on the trigger syntax for an 'endpoint' task.

Could someone with more knowledge than me in these matters please advise as to whether this could possibly be a problem, else close this as a non-issue?

#### Minimal working example

```ini
[scheduling]
    [[dependencies]]
        graph = """
            X => Y:any-alphanumeric-text-at-least
        """

```
passes (even strict) validation, with task Y being triggered normally & succeeding. Note: clearly a graph string of ``Y:any-alphanumeric-text-at-least => X`` would fail validation.

**Comment by hjoliver** [comment]
Strongly affected (/made irrelevant?) by cylc/cylc-admin#129

**Comment by oliver-sanders** [comment]
I think this should be closed by #4343
